sources/Tools/Visualiser.py

```diff
@@ -126,26 +125,51 @@ def _process_bar_data(raw_data, head):
             "left": [(stat[index] == "Left").sum() for index in range(4)],
         }
 
-    def _histogram_visualizer(self, head):
-        logging.warning(
-            f"{inspect.currentframe().f_code.co_name}:Need to be refactor - No hard coded data please"
-        )
-        hands = self._process_bar_data(self.raw_data, head)
-        houses = set(self.raw_data.loc[:, "Hogwarts House"])
-        x = np.arange(len(houses))
-        width = 0.35
-        fig, ax = plt.subplots()
-        # print(hands)
-        rects1 = ax.bar(x - width / 2, hands["right"], width, label="Right")
-        rects2 = ax.bar(x + width / 2, hands["left"], width, label="Left")
-        ax.set_xticks(x)
-        ax.set_xticklabels(houses)
-        ax.legend()
-        self._auto_label(ax, rects1)
-        self._auto_label(ax, rects2)
-        fig.tight_layout()
+    @staticmethod
+    def _pick_by_elements(raw_data, filters, column_filter, feature):
+        return [raw_data.loc[raw_data[column_filter] == element, feature] for element in filters]
+
+    @staticmethod
+    def _create_histogram(
+        raw_data: list, title="Histogram", kde_label=None, xlabel="x", ylabel="y", yticks=None
+    ):
+        fig = plt.figure()
+        for index, data in enumerate(raw_data):
+            plt.hist(data, density=True, bins=30, alpha=0.5)
+            if kde_label is not None:
+                mn, mx = plt.xlim()
+                plt.xlim(mn, mx)
+                kde_xs = np.linspace(mn, mx, 301)
+                kde = st.gaussian_kde(data)
+                plt.plot(kde_xs, kde.pdf(kde_xs), label=kde_label[index])
+        plt.title(title)
+        plt.legend(kde_label, loc="upper left")
+        plt.xlabel(xlabel)
+        plt.ylabel(ylabel)
+        plt.yticks([] if yticks is None else yticks)
+        plt.close()
         return fig
 
+    def _histogram_visualizer(self, header):
+        figures = [self._text_page()]
+        for feature in header:
+            try:

```

Comment:
when could it crash, i think the try except is mostly used to catch ONE excpetions here is like a mega trash for the 2 funcs maybe rethinks it or call me for a debate